api/v3/Contribution.php

```diff
@@ -585,14 +585,11 @@ function _civicrm_api3_contribution_completetransaction_spec(&$params) {
 function civicrm_api3_contribution_repeattransaction($params) {
   civicrm_api3_verify_one_mandatory($params, NULL, ['contribution_recur_id', 'original_contribution_id']);
   if (empty($params['original_contribution_id'])) {
-    //  CRM-19873 call with test mode.
-    $params['original_contribution_id'] = civicrm_api3('contribution', 'getvalue', [
-      'return' => 'id',
-      'contribution_status_id' => ['IN' => ['Completed']],
-      'contribution_recur_id' => $params['contribution_recur_id'],
-      'contribution_test' => CRM_Core_DAO::getFieldValue('CRM_Contribute_DAO_ContributionRecur', $params['contribution_recur_id'], 'is_test'),
-      'options' => ['limit' => 1, 'sort' => 'id DESC'],
-    ]);
+    $templateContribution = CRM_Contribute_BAO_ContributionRecur::getTemplateContribution($params['contribution_recur_id']);
+    if (empty($templateContribution)) {
+      throw new CiviCRM_API3_Exception('Contribution.repeattransaction failed to get original_contribution_id for recur with ID: ' . $params['contribution_recur_id']);
```

Comment:
can you change to `API_Exception` to make it APIv4 friendly too

Reply:
Actually I think within apiv3 this is correct. 

API_Exception is a bit of a pain in a way - ideally we could catch 'all CiviCRM extensions' with one catch type but API_Exception doesn't extend any other CiviCRM exception types - I'm guessing that was before it was handed over to CT @colemanw ?

Reply:
@monishdeb This code is within an API3 function so it is unlikely to ever be called by API4?  There is quite a bit within the function which would probably be cleaned up before creating a "repeattransaction" for API4 - see eg. https://github.com/civicrm/civicrm-core/pull/22488 for a start.